api/src/main/java/org/pmiops/workbench/api/CohortsController.java

```diff
@@ -149,8 +160,16 @@ public Cohort apply(org.pmiops.workbench.db.model.Cohort cohort) {
     }
     Timestamp now = new Timestamp(clock.instant().toEpochMilli());
     dbCohort.setLastModifiedTime(now);
-    // TODO: add version, check it here
-    dbCohort = cohortDao.save(dbCohort);
+    try {
+      // The version asserted on save is the same as the one we read via
+      // getRequired() above, see RW-215 for details.
+      dbCohort = cohortDao.save(dbCohort);
+    } catch (OptimisticLockException e) {
+      log.log(Level.WARNING, "version conflict for cohort update", e);
+      throw new ConflictException("Failed due to concurrent cohort modification");
+    }
+    // TODO: Understand why CrudRepository does not return an incremented version.
+    dbCohort.setVersion(dbCohort.getVersion() + 1);
```

Comment:
So if you don't do this, dbCohort will have the old version?

Reply:
I think this SO post may be discussing this sort of issue:

https://stackoverflow.com/questions/33825429/spring-data-refresh-entity-after-manual-backend-query-update

It sounds like an alternative is to inject @PersistenceContext EntityManager into this class, and call refresh(dbCohort) here.